consensus/src/main/java/org/ethereum/beacon/consensus/util/CachingBeaconChainSpec.java

```diff
@@ -25,35 +31,72 @@
       new LRUCache<>(1024);
   private final LRUCache<Object, Hash32> signedRootCache =
       new LRUCache<>(1024);
+  private final LRUCache<SlotNumber, List<ShardCommittee>> crosslinkCommitteesCache =
+      new LRUCache<>(128);
+  private final LRUCache<EpochNumber, List<ValidatorIndex>> activeValidatorsCache =
+      new LRUCache<>(32);
+  private final LRUCache<EpochNumber, Gwei> totalBalanceCache =
+      new LRUCache<>(32);
   private ValidatorIndex maxCachedIndex = ValidatorIndex.ZERO;
   private final Map<BLSPubkey, ValidatorIndex> pubkeyToIndexCache = new ConcurrentHashMap<>();
 
+  private final boolean cacheEnabled;
+
   public CachingBeaconChainSpec(
       SpecConstants constants,
       Function<BytesValue, Hash32> hashFunction,
-      ObjectHasher<Hash32> objectHasher) {
-    super(constants, hashFunction, objectHasher);
+      ObjectHasher<Hash32> objectHasher,
+      boolean blsVerify,
+      boolean blsVerifyProofOfPossession,
+      boolean cacheEnabled) {
+    super(constants, hashFunction, objectHasher, blsVerify, blsVerifyProofOfPossession);
+    this.cacheEnabled = cacheEnabled;
+  }
+
+  public CachingBeaconChainSpec(
+      SpecConstants constants,
+      Function<BytesValue, Hash32> hashFunction,
+      ObjectHasher<Hash32> objectHasher,
+      boolean blsVerify,
+      boolean blsVerifyProofOfPossession) {
+    this(constants, hashFunction, objectHasher, blsVerify, blsVerifyProofOfPossession, true);
   }
 
   @Override
   public List<UInt64> get_permuted_list(List<? extends UInt64> indices, Bytes32 seed) {
+    if (!cacheEnabled) {
+      return super.get_permuted_list(indices, seed);
+    }
+
     return shufflerCache.get(
         Pair.with(indices, seed),
         k -> super.get_permuted_list(k.getValue0(), k.getValue1()));
   }
 
   @Override
   public Hash32 hash_tree_root(Object object) {
+    if (!cacheEnabled) {
+      return super.hash_tree_root(object);
+    }
+
```

Comment:
Just an idea: instead of duplicating super-calls in each method, m.b. it makes sense to put a 'dummy' `LRUCache` instance if caching disabled?

Reply:
I like this idea. Will give it a shot.